Add basic documentation.

mgautierfr opened the pull request:
- Basic usage. Mainly taken from https://wiki.openzim.org/wiki/Libzim with (a lot of) adaptation.
- Transition guide from libzim6 to libzim7.

This is a basic documentation and doesn't cover all features of libzim.

Fix #608 

Generated doc can be seen here : https://libzim--611.org.readthedocs.build/en/611/

maneeshpm commented on docs/usage.rst:
```diff
@@ -0,0 +1,168 @@
+Searching and creating part are not. You have to serialize access to the class yourself.
+
+The main class, which accesses a archive is |Archive|.
+It has actually a reference to a implementation, so that copies of the class just references the same file.
```
reference to *an* implementation

maneeshpm reviewed (changes requested):
Just some minor typo errors and a few suggestions. Good one!

rgaudin reviewed (approved):
Great addition

veloman-yunkan reviewed (approved):
Fixing typos & grammar and improving some pieces of the (new) documentation will unnecessarily delay merging this PR. It is a good start, we can merge it as is and then have someone refine it.

maneeshpm reviewed (approved):
I agree with @veloman-yunkan, we can refine it later.

mgautierfr commented on docs/6to7.rst:
```diff
@@ -0,0 +1,371 @@
+
+In libzim7 you search starting from a |Searcher|.
+
+ .. code-block:: c++
```
I've removed all includes for consistency.

mgautierfr commented on docs/6to7.rst:
```diff
@@ -0,0 +1,371 @@
+
+ .. code-block:: c++
+
+    #include <zim/archive.h>
```
I've removed all includes for consistency.

mgautierfr commented:
Rebased on master and I've fixed the few typos.
Feel free to create PR to improve the documentation.

Deb packaging CI is failing but nothing related to this PR. I'm merging.
